KhushSharma_KaprekarConstantAssignmentz.java

```diff
@@ -0,0 +1,81 @@
+import java.util.Arrays;
+import java.util.Scanner;
+import java.util.InputMismatchException;
+
+class KaprekarConstant {
+
+	private static int findKaprekarRoutine(int number) {
+
+		int iterations = 0;
+		int numberOfDigits = (Integer.toString(number)).length();
+
+		int[] arrayOfDigits = new int[numberOfDigits];
+
+		while (number != 6174) {
+			for (int i = numberOfDigits - 1; i >= 0; i--) {
+				int digit = number % 10;
+				arrayOfDigits[i] = digit;
+				number /= 10;
+			}
+
+			if (arrayOfDigits[0] == arrayOfDigits[1] && arrayOfDigits[1] == arrayOfDigits[2]
+					&& arrayOfDigits[2] == arrayOfDigits[3])
+				throw new InputMismatchException();
+
+			Arrays.sort(arrayOfDigits);
+
+			int ascendingNumber = 0;
+			int descendingNumber = 0;
+
+			for (int i = 0; i < numberOfDigits; i++) {
```

Comment:
This loop code is a different functionality. So, It can be converted into a separate function.